src/main/java/core/basesyntax/filereader/FileReaderImpl.java

```
@@ -0,0 +1,22 @@
+package core.basesyntax.filereader;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FileReaderImpl implements FileReaderService {
+
+    @Override
+    public List<String> read(String path) {
+        List<String> stringLines = new ArrayList<>();
+
+        try {
+            stringLines = Files.readAllLines(Path.of(path));
+        } catch (IOException e) {
+            throw new RuntimeException(e);

```

This violates checklist item #15: 'Throw RuntimeException with informative messages instead of printing stack traces when I/O errors occur (e.g., file not found).' The exception thrown here does not include an informative message for the user.